tesseract_collision/test/collision_sphere_sphere_unit.cpp

```diff
@@ -158,19 +159,171 @@ void runTest(tesseract::DiscreteContactManagerBase& checker)
+  EXPECT_NEAR(result_vector[0].nearest_points[0][2], result_vector[0].nearest_points[1][2], 0.001);
+  EXPECT_NEAR(result_vector[0].normal[0], idx[2] * 1.0, 0.001);
+  EXPECT_NEAR(result_vector[0].normal[1], idx[2] * 0.0, 0.001);
+  EXPECT_NEAR(result_vector[0].normal[2], idx[2] * 0.0, 0.001);
+
+  //////////////////////////////////////////////////////////////////////
+  // Test when object is in collision (Closest Feature Vertex to Vertex)
+  //////////////////////////////////////////////////////////////////////
+  location["sphere1_link"] = Eigen::Isometry3d::Identity();
+  location["sphere1_link"].translation()(1) = 0.2;
+  checker.setCollisionObjectsTransform(location);
+
+  // Perform collision check
+  result.clear();
+  result_vector.clear();
+
+  checker.contactTest(result);
+  tesseract::moveContactResultsMapToContactResultsVector(result, result_vector);
+
+  EXPECT_TRUE(!result_vector.empty());
+  EXPECT_NEAR(result_vector[0].distance, -0.25, 0.005);
+
+  idx = { 0, 1, 1 };
+  if (result_vector[0].link_names[0] != "sphere_link")
+    idx = { 1, 0, -1 };
+
+  EXPECT_NEAR(result_vector[0].nearest_points[idx[0]][1], 0.25, 0.005);
+  EXPECT_NEAR(result_vector[0].nearest_points[idx[1]][1], -0.05, 0.005);
+  EXPECT_NEAR(result_vector[0].nearest_points[0][0], result_vector[0].nearest_points[1][0], 0.005);
+  EXPECT_NEAR(result_vector[0].nearest_points[0][2], result_vector[0].nearest_points[1][2], 0.005);
+
+  EXPECT_NEAR(result_vector[0].normal[0], idx[2] * 1.0, 0.001);
+  EXPECT_NEAR(result_vector[0].normal[1], idx[2] * 0.0, 0.001);
+  EXPECT_NEAR(result_vector[0].normal[2], idx[2] * 0.0, 0.001);
+
+  ////////////////////////////////////////////////
+  // Test object is out side the contact distance
+  ////////////////////////////////////////////////
+  location["sphere1_link"].translation() = Eigen::Vector3d(0, 1, 0);
+  result.clear();
+  result_vector.clear();
+  checker.setCollisionObjectsTransform(location);
+
+  checker.contactTest(result);
+  tesseract::moveContactResultsMapToContactResultsVector(result, result_vector);
+
+  EXPECT_TRUE(result_vector.empty());
+
+  /////////////////////////////////////////////////////////////////////////////
+  // Test object inside the contact distance (Closest Feature Vertex to Vertex)
+  /////////////////////////////////////////////////////////////////////////////
+  result.clear();
+  result_vector.clear();
+  req.contact_distance = 0.51;
+  checker.setContactRequest(req);
+
+  // The closest feature of the mesh should be edge to edge
+  checker.contactTest(result);
+  tesseract::moveContactResultsMapToContactResultsVector(result, result_vector);
+
+  EXPECT_TRUE(!result_vector.empty());
+  EXPECT_NEAR(result_vector[0].distance, 0.5, 0.001);
+
+  idx = { 0, 1, 1 };
+  if (result_vector[0].link_names[0] != "sphere_link")
+    idx = { 1, 0, -1 };
+
+  EXPECT_NEAR(result_vector[0].nearest_points[idx[0]][1], 0.25, 0.001);
```

Comment:
This is not a specific issue, but just an idea: Is it possible to manually compute what the "right answer" should be based on pure geometry and then confirm that the checker gets it right? This is opposed to hard-coding the results. This is not likely to change a lot, but makes the tests more generally verifiable.

Reply:
Good point

Reply:
I will create an issue to not hold up this PR